aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/SynchronousNonBlockingStepTest.java

```diff
@@ -0,0 +1,245 @@
+package org.jenkinsci.plugins.workflow.steps;
+
+import static org.junit.Assert.assertTrue;
+import hudson.model.Result;
+import hudson.model.TaskListener;
+import hudson.model.Run;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowExecution;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepNode;
+import org.jenkinsci.plugins.workflow.graph.FlowGraphWalker;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import com.google.inject.Inject;
+
+public class SynchronousNonBlockingStepTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void basicNonBlockingStep() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node {\n" +
+            "echo 'First message'\n" +
+            "syncnonblocking 'wait'\n" +
+            "echo 'Second message'\n" +
+        "}"));
+        WorkflowRun b = p.scheduleBuild2(0).getStartCondition().get();
+
+        // Wait for syncnonblocking to be started
+        System.out.println("Waiting to syncnonblocking to start...");
+        SynchronousNonBlockingStep.waitForStart("wait", b);
+
+        // At this point the execution is paused inside the synchronous non-blocking step
+        // Check for FlowNode created
+        FlowGraphWalker walker = new FlowGraphWalker(b.getExecution());
+        boolean found = false;
+        for (FlowNode n = walker.next(); n != null; n = walker.next()) {
+            if (n instanceof StepNode) {
+                found = true;
```

Comment:
:ant: this would pass if only the `echo` step had run. You need to verify the `StepDescriptor`!